### DISCUSSION TITLE
How to define custom GraphQL type?

### DISCUSSION THREAD

**Discussion by iBobik** [contember/admin#642] [Support]
Where to define that GraphQL type?

```
publishedAt = def.dateTimeColumn().typeAlias('CustomDateTime')
```
https://docs.contember.com/reference/engine/schema/columns#changing-column-type

I want to use jsonColumn and specify how that object should look like, so Contember will enforce it and TypeScript on the frontend will know what to expect.

**Comment by matej21** [comment]
Hi,

Unfortunately, Contember does not currently support defining the schema of a type alias, particularly for a jsonColumn. What this means is that it will only produce a GraphQL scalar alias that behaves the same as the original type.

However, this setup does allow you to treat the type alias differently on the client-side using your GraphQL client. For instance, you can use the ["scalars"](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript#scalars) config in GraphQL Code Generator to achieve this.
